pkg/scalers/azure_servicebus_scaler.go

```diff
@@ -170,11 +158,26 @@ func parseAzureServiceBusMetadata(config *ScalerConfig, logger logr.Logger) (*az
 			return nil, fmt.Errorf("no connection setting given")
 		}
 	case kedav1alpha1.PodIdentityProviderAzure, kedav1alpha1.PodIdentityProviderAzureWorkload:
-		if val, ok := config.TriggerMetadata["namespace"]; ok {
-			meta.namespace = val
+		if val, ok := config.TriggerMetadata["fullyQualifiedNamespace"]; ok {
+			meta.fullyQualifiedNamespace = val
 		} else {
-			return nil, fmt.Errorf("namespace is required when using pod identity")
+			if val, ok := config.TriggerMetadata["namespace"]; ok {
+				logger.Info("Warning: namespace field has been deprecated in favor of fullyQualifiedNamespace")
```

Comment:
Frankly, I think we should keep the current configuration and determine this ourselves as we have all the information we need (cloud + namespace name) which makes it simpler for end-users IMO.

The majority will use public cloud where just using the namespace name is enough.

Reply:
I'm not sure if it's worth because the underlying SDK doesn't have the endpoint information, it's other SDK who has that info. We can have that couple for the moment and see what to do in the future, keeping the current behavior as default.
I'll update the code to not add any extra value

Reply:
Thank you! I think it's easy to do this by using a simple map between cloud and the endpoint suffix.

Reply:
done!

Reply:
Done, PTAL